### ISSUE TITLE
Is there a way to publish from a specific directory?

### ISSUE THREAD

**Issue by jrschumacher** [intuit/auto#1423]
I use a tool like [pmsandhu/copy-pkg-json-webpack-plugin](https://github.com/pmsandhu/copy-pkg-json-webpack-plugin) which allows me to strip out dependencies and devDependencies since the archives are built. 

Is there a way to get auto to change directory before publish?

**Comment by hipstersmoothie** [comment]
Could I see the project? Auto is typically run from the root, never tried just a folder

**Comment by jrschumacher** [comment]
@hipstersmoothie I'll have to put something together. I haven't implemented it into my current project yet. 

I'm also fine writing a script to remove package.json before publish and restoring it after. I just don't want it to commit the truncated package.json. 

Are there hooks I can tie into via autorc?

**Comment by hipstersmoothie** [comment]
You can use the [exec plugin](https://intuit.github.io/auto/docs/generated/exec) to tap all of the [available hook](https://intuit.github.io/auto/docs/plugins/hook-api-docs) but there might be a simpler way. i'd just have to see a sample repo

**Comment by jrschumacher** [comment]
Gotcha. Looking at the code, the issue is that NPM uses `npm` git integration so that won't work (since I would need to chdir to the dist directory.)

**Comment by jdalrymple** [comment]
@jrschumacher Did you manage to figure out a good way of doing this? I know there are ways to do this [manually](https://stackoverflow.com/a/48460033/1812936), but through auto :thinking:

**Comment by sebko** [comment]
I'm also running into this issue. We're deploying our component library from the `/dist` folder to avoid our imports looking like this:

```
import Button from `<component_library_name>/dist/Button`
```

We'd like them to look like this instead:
```
import Button from `<component_library_name>/Button`
```

To get around this issue, we run the following code in our CI/CD environment:
```
yarn dist
yarn run copy-package-json.js // Copies package.json file to `/dist`
cd dist // Ensure auto command is run within the dist folder
npx auto shipit
```

This issue with the code above is that the `git` commands fail because the `/dist` folder is `.gitignored`.

**Comment by intuit-svc** [comment]
:rocket: Issue was released in [`v10.33.0`](https://github.com/intuit/auto/releases/tag/v10.33.0) :rocket:
